DatadogCore/Tests/Datadog/Core/Upload/DataUploaderTests.swift

```diff
@@ -10,20 +10,18 @@ import TestUtilities
 @testable import DatadogCore
 
 class DataUploaderTests: XCTestCase {
-    func testWhenUploadCompletesWithSuccess_itReturnsExpectedUploadStatus() throws {
+    // swiftlint:disable opening_brace
+    func testGivenValidRequest_whenUploadCompletesWithStatusCode_itReturnsUploadStatus() throws {
         // Given
         let randomResponse: HTTPURLResponse = .mockResponseWith(statusCode: (100...599).randomElement()!)
-        let randomRequestIDOrNil: String? = Bool.random() ? .mockRandom() : nil
-        let requestIDHeaderOrNil: URLRequestBuilder.HTTPHeader? = randomRequestIDOrNil.flatMap { randomRequestID in
-                .init(field: URLRequestBuilder.HTTPHeader.ddRequestIDHeaderField, value: { randomRequestID })
-        }
-
-        let server = ServerMock(delivery: .success(response: randomResponse))
```

Comment:
💡 A common pattern in this PR - we can now replace most usages of `ServerMock` with `HTTPClientMock`. The latter doesn't depend on mocking `URLProtocol` and we can use multiple instances of `HTTPClientMock` in single test.